lazyflow/operator.py

```diff
@@ -30,6 +30,28 @@
 from lazyflow.slot import InputSlot, OutputSlot, Slot
 
 
+class _Transaction:
```

Comment:
This class could probably be moved into `Operator`:
```python
class Operator:
    def __init__(self):
        self._deferred_callbacks = None

    @contextmanager
    def transaction(self):
        assert self._deferred_callbacks is None
        self._deferred_callbacks = []
        yield
        callbacks = self._deferred_callbacks
        self._deferred_callbacks = None
        for cb in callbacks:
            cb()


class Slot:
    def _maybe_call_within_transaction(self, fn):
        op = getattr(self, "operator", None)
        callbacks = getattr(op, "_deferred_callbacks", None)
        if callbacks is None:
            fn()
        elif fn not in callbacks:
            callbacks.append(fn)
```

Reply:
I will move this class under Operator namespace, but I am reluctant putting yet another responsibility fully inside `Operator`.
> ```python
>         if callbacks is None:
>             fn()
>         elif fn not in callbacks:
>             callbacks.append(fn)
> ```
deduplication of callbacks done inside transaction by set

Reply:
> I will move this class under Operator namespace, but I am reluctant putting yet another responsibility fully inside `Operator`.

Agreed. I suggested the change because `_Transaction` is small and mutually depends on `Operator`.

> deduplication of callbacks done inside transaction by set

This suggestion was made to preserve callback execution order: `set` doesn't do that, and `list` is simpler/faster than `dict` on small sizes.